packages/flutter/lib/src/foundation/binding.dart

```diff
@@ -40,6 +61,167 @@ abstract class BindingBase {
+  void registerSignalServiceExtension({ 
+    @required String name,
+    @required VoidCallback callback
+  }) {
+    assert(name != null);
+    assert(callback != null);
+    registerServiceExtension(
+      name: name,
+      callback: (Map<String, String> parameters) async {
+        callback();
+        return <String, dynamic>{};
+      }
+    );
+  }
+
+  /// Registers a service extension method with the given name (full
+  /// name "ext.flutter.name"), which takes a single argument
+  /// "enabled" which can have the value "true" or the value "false"
+  /// or can be omitted to read the current value. (Any value other
+  /// than "true" is considered equivalent to "false". Other arguments
+  /// are ignored.)
+  ///
+  /// Invokes the `getter` callback to obtain the value when
+  /// responding to the service extension method being invoked.
+  ///
+  /// Invokes the `setter` callback with the new value when the
+  /// service extension method is invoked with a new value.
+  void registerBoolServiceExtension({ 
+    String name,
+    @required ValueGetter<bool> getter,
+    @required ValueChanged<bool> setter
+  }) {
+    assert(name != null);
+    assert(getter != null);
+    assert(setter != null);
+    registerServiceExtension(
+      name: name, 
+      callback: (Map<String, String> parameters) async {
+        if (parameters.containsKey('enabled'))
+          setter(parameters['enabled'] == 'true');
+        return <String, dynamic>{ 'enabled': getter() };
+      }
+    );
+  }
+
+  /// Registers a service extension method with the given name (full
+  /// name "ext.flutter.name"), which takes a single argument with the
+  /// same name as the method which, if present, must have a value
+  /// that can be parsed by [double.parse], and can be omitted to read
+  /// the current value. (Other arguments are ignored.)
+  ///
+  /// Invokes the `getter` callback to obtain the value when
+  /// responding to the service extension method being invoked.
+  ///
+  /// Invokes the `setter` callback with the new value when the
+  /// service extension method is invoked with a new value.
+  void registerNumericServiceExtension({ 
+    @required String name,
+    @required ValueGetter<double> getter,
+    @required ValueChanged<double> setter
+  }) {
+    assert(name != null);
+    assert(getter != null);
+    assert(setter != null);
+    registerServiceExtension(
+      name: name,
+      callback: (Map<String, String> parameters) async {
+      if (parameters.containsKey(name))
+        setter(double.parse(parameters[name]));
+      return <String, dynamic>{ name: getter() };
+    });
+  }
+
+  /// Registers a service extension method with the given name (full
+  /// name "ext.flutter.name"). The given callback is invoked when the
+  /// extension method is called. The callback must return a [Future]
+  /// that either eventually completes to a return value in the form
+  /// of a name/value map, or fails. In case of failure, the failure
```

Comment:
I'd say "in the form that can be encoded as a JSON object using `dart:convert`" to emphasize that we do fully support JSON.